l10n_th_tin_service/models/res_partner.py

```diff
@@ -0,0 +1,196 @@
+        "https://rdws.rd.go.th/serviceRD3/checktinpinservice.asmx?wsdl"
+    )
+    vat_web_service_url = "https://rdws.rd.go.th/serviceRD3/vatserviceRD3.asmx?wsdl"
+
+    @staticmethod
+    def check_rd_tin_service(tin):
+        """Return bool after verifiying Tax Identification Number (TIN)
+           or Personal Identification Number (PIN)
+           by using Revenue Department's web service to prevent forging.
+           :param tin: a string for TIN or PIN
+        """
+
+        sess = Session()
+        mod_dir = path.dirname(path.realpath(__file__))
+        cert_path = str(Path(mod_dir).parents[0]) + "/static/cert/adhq1_ADHQ5.cer"
+        sess.verify = cert_path
+        transp = Transport(session=sess)
+        try:
+            cl = Client(ResPartner.tin_web_service_url, transport=transp)
+        except requests.exceptions.SSLError:
+            _logger.log(INFO, "Fall back to unverifed HTTPS request.")
+            sess.verify = False
+            transp = Transport(session=sess)
+            cl = Client(ResPartner.tin_web_service_url, transport=transp)
+        result = cl.service.ServiceTIN("anonymous", "anonymous", tin)
+        res_ord_dict = helpers.serialize_object(result)
+        _logger.log(DEBUG, pprint.pformat(res_ord_dict))
+        return res_ord_dict["vIsExist"] is not None
+
+    @staticmethod
+    def get_info_rd_vat_service(tin, branch=0):
+        """Return ordered dict with necessary result from
+           Revenue Department's web service.
+           :param tin: a string for TIN or PIN
+           :param branch: one digit of branch number
+        """
+        branch = int(branch)
+        sess = Session()
+        mod_dir = path.dirname(path.realpath(__file__))
+        cert_path = str(Path(mod_dir).parents[0]) + "/static/cert/adhq1_ADHQ5.cer"
+        sess.verify = cert_path
+        transp = Transport(session=sess)
+        try:
+            cl = Client(ResPartner.vat_web_service_url, transport=transp)
+        except requests.exceptions.SSLError:
+            _logger.log(INFO, "Fall back to unverifed HTTPS request.")
+            sess.verify = False
+            transp = Transport(session=sess)
+            cl = Client(ResPartner.vat_web_service_url, transport=transp)
+        result = cl.service.Service(
+            "anonymous",
+            "anonymous",
+            TIN=tin,
+            ProvinceCode=0,
+            BranchNumber=branch,
+            AmphurCode=0,
+        )
+        odata = helpers.serialize_object(result)
+        _logger.log(DEBUG, pprint.pformat(odata))
+        data = OrderedDict()
+        if odata["vmsgerr"] is None:
+            for key, value in odata.items():
+                if (
+                    value is None
+                    or value["anyType"][0] == "-"
+                    or key in {"vNID", "vBusinessFirstDate"}
+                ):
+                    continue
+                data[key] = value["anyType"][0]
+            return data
+        return False
```

Comment:
Please use this pattern to improve code readability.

```suggestion
        if odata["vmsgerr"]:
            return False
        for key, value in odata.items():
            if (value and value["anyType"][0] != "-" and key not in {"vNID", "vBusinessFirstDate"} ):
                data[key] = value["anyType"][0]
        return data
```